scripts/ram.sh

```diff
@@ -14,42 +14,43 @@ set -o nounset
 #
 # VARIABLES :
 #
-#Des Couleurs 
-# 
-bold=$(tput bold) 
-underline=$(tput sgr 0 1) 
-reset=$(tput sgr0) 
-purple=$(tput setaf 171) 
-red=$(tput setaf 1) 
-green=$(tput setaf 76) 
-tan=$(tput setaf 3) 
-blue=$(tput setaf 38) 
-# 
+#Des Couleurs
+#
+bold=$(tput bold)
+underline=$(tput sgr 0 1)
+reset=$(tput sgr0)
+purple=$(tput setaf 171)
+red=$(tput setaf 1)
+green=$(tput setaf 76)
+tan=$(tput setaf 3)
+blue=$(tput setaf 38)
+#
 # En-têtes et Logs
-# 
-e_header() { printf "\n${bold}${purple}==========  %s  ==========${reset}\n" "$@"  
-} 
-e_arrow() { printf "➜ $@\n" 
-} 
-e_success() { printf "${green}✔ %s${reset}\n" "$@" 
-} 
-e_error() { printf "${red}✖ %s${reset}\n" "$@" 
-} 
-e_warning() { printf "${tan}➜ %s${reset}\n" "$@" 
-} 
-e_underline() { printf "${underline}${bold}%s${reset}\n" "$@" 
-} 
-e_bold() { printf "${bold}%s${reset}\n" "$@" 
-} 
-e_note() { printf "${underline}${bold}${blue}Note:${reset}  ${blue}%s${reset}\n" "$@" 
-} 
+#
+e_header() { printf "\n${bold}${purple}==========  %s  ==========${reset}\n" "$@"; }
+e_arrow() { printf "➜ $@\n"; }
+e_success() { printf "${green}✔ %s${reset}\n" "$@"; }
+e_error() { printf "${red}✖ %s${reset}\n" "$@"; }
+e_warning() { printf "${tan}➜ %s${reset}\n" "$@"; }
+e_underline() { printf "${underline}${bold}%s${reset}\n" "$@"; }
+e_bold() { printf "${bold}%s${reset}\n" "$@"; }
+e_note() { printf "${underline}${bold}${blue}Note:${reset}  ${blue}%s${reset}\n" "$@"; }
 #
 # FONCTIONS :
 #
 # source /etc/skel/.scripts/functions.sh
 #
-# LE SCRIPT PROPREMENT DIT : 
+# LE SCRIPT PROPREMENT DIT :
 #
 
-ps aux | awk '{print $6 "\t" $11}' | sort -rn | head -25 | awk '{print $1/1024 " MB\t\t" $2}'
+#ps aux | awk '{print $6 "\t" $11}' | sort -rn | head -25 | awk '{print $1/1024 " MB\t\t" $2}'
+
+awk "
+	{
+		M=\$1/1024
+		if(NR<25 && M>1){
+			printf(\"%'7dMB  %s\\n\", M, \$2)
+		}
+	}
+" <<< "$(ps ax -o rss= -o comm= --sort -rss)"
```

Comment:
The output is similar, just cleaner and traditional. I hope you like it! It was a pain to do. lol

The `NR<25` part is for the maximum number of lines to display, akin to the `head` command, so you can change the `25` to whatever integer you want.

The `M>1` means that processes whose `rss` (physical memory usage) is less than 1MB will not be displayed. Usually small processes like that aren't of interest with output like this. It also saves programming `awk` to display the numbers properly, with the correct notation and equations.

As you can see, `ps` itself is incredibly flexible. Here, I've specified which columns I want, that I don't want a header, and that they are to be sorted by `rss`, from highest to lowest (`-rss`).

I've removed the `u` part (previously `aux`) because `ps` refused to otherwise work without spitting out errors. The `u`, if I remember correctly, means that it displays processes for either the specified user, or, if none is given, the current user. However, `a` is provided, meaning all processes are displayed, and I think that means regardless of ownership.